### DISCUSSION TITLE
"message":"No API key found in request","hint":"No `apikey` request header or url param was found."

### DISCUSSION THREAD

**Discussion by Arisz7** [supabase/supabase#16241] [General]
Hello i use to have a deploy project that it was working just fine it is about a website that is intended to be a clone from spotify in which one you can reproduce songs. The problem comes when i tried to re deploy it again after a restart the project from being paused i was able to log in and create new accounts when i tried to reproduce a song i was getting the error that i mentioned before in the discussion title, i dont have any clue about it cause the code it was working just fine before i commited the mistake after i delete the project. Can some one help me with this situation i tried to post my question in other platforms to get help but none of them seems to want to help me, and yes i have all the keys in order for supabase to be running i provided to vercel after set up the deployment and still not working

**Comment by parsasi** [comment]
I'm facing the same problem on a deployment.
The request returns:
```json
{"message":"No API key found in request","hint":"No `apikey` request header or url param was found."}
```
This is while still on the superbase URL, before getting redirected to my domain. I can see the URL hash parameters in the URL too.

**Reply by lissy93** [reply]
Hey @parsasi 
Did you eve find a solution to this? I've got the exact same issue, and am struggling to figure it out
Any hints would be much appreciated 💙

**Reply by lissy93** [reply]
Spent a good couple of hours stuck on this. But typically the moment I asked for help I saw the issue!
Posting here in case anyone else comes across a similar problem.

I just needed to include the protocol in the Site URL (under Auth --> Site Configuration). 

Without `https://` I was seeing the `No apikey request header or url param was found` error, but with it, it works fine :)

![Screenshot from 2023-11-29 17-09-47](https://github.com/supabase/supabase/assets/1862727/e181469c-d8c8-4ef5-98c5-5a089624576d)

**Reply by fafa16** [reply]
hello @Lissy93 could you please explain what you mean by "I just needed to include the protocol in the Site URL (under Auth --> Site Configuration)."? I have an url setup here but what do I need to do?

thank you by advance..
Fab

**Reply by audiojak** [reply]
Thanks @Lissy93! That worked.

**Reply by rosalynw** [reply]
That worked! Thank you so much

**Comment by jgruber** [comment]
I am using the free tier and the only project URL I have is the project URL generated by supabase.com. 

ie: https://xxxxxxxxxxxxxxxxxxx.supabase.co

What should be in the Site Url setting for authentication to get any of the confirmation URLs to be correct. Right now all I see is all of the query parameters matching parameters you would get from an access token request:

https://xxxxxxxxxxxxx.supabase.co/#access_token=XXXXXXXX&expires_at=1712757484&expires_in=3600&refresh_token=XXXXXXXXXX&token_type=bearer&type=invite

and still get the same error response:

{"message":"No API key found in request","hint":"No `apikey` request header or url param was found."}

**Reply by jfitzsimmons2** [reply]
FYI - this was happening to me when I had RLS enabled but didn't have a proper RLS policy set up for the operation I was trying to perform. e.g. I was trying to write to a table where I had policies for SELECT and DELETE but not INSERT.

**Reply by k61b** [reply]
@jfitzsimmons2 You are a wonderful person. Thank you very much.

**Comment by HyperMonkey3D** [comment]
Hey, in case anyone else is having this issue and the solutions above don't work, just make sure that you have the right table name in your queries. Silly mistake, I know, but I spent a good chunk of time until I realized that was my issue.

**Comment by leonaburime** [comment]
I had this same problem. Turns out I was setting the session multiple times with auth and refresh token. I wasnt checking to see if this call was giving me an error

```
await this.supabase.auth.setSession({
          access_token: authData.session.access_token,
          refresh_token: authData.session.refresh_token,
        });
```
        
In fact doin the above command multiple times was causing me to use that refresh token too early and then when trying to use it again I couldnt refresh and make authenticated calls

**Comment by shahbaz-khan-hub** [comment]
hey i have added this url `https:localhost:1234/`but it still wont work at all it also says `captch AuthError`

**Comment by LDK1009** [comment]
I encountered a problem during bulk insert.
The solution was a single row insert.

**Comment by richardverheyen** [comment]
This error was throwing for me because I didn't know that I needed to at Table-level policies as well as RLS policies, and I wasn't paying close enough attention to the Roles that my frontend requests were being sent on behalf of.

**Reply by schmidpatric** [reply]
exactly that! Had to enable a read policy in https://supabase.com/dashboard/project/xxx/auth/policies

**Comment by wassimrahali** [comment]
you need to put api key as header it worked for me after 3hours XD
<img width="913" height="240" alt="image" src="https://github.com/user-attachments/assets/ac25a639-49e1-4904-8d02-3c760db05125" />
